docs/library/utime.rst

```diff
@@ -28,8 +28,8 @@ by following means:
   in such case).
 
 If actual calendar time is not maintained with a system/MicroPython RTC,
-functions below which require reference to current absolute time may
-behave not as expected.
+functions below which require reference to current absolute time may not
```

Comment:
I can assure you that formulations like above were purposedly (purposely?) crafted in a way to minimize ambiguity based on the formal structure of English as a *language* (in a formal sense, even if it's still a highly ambiguous natural language).

The most obvious (almost the only possible) way to parse that phrase is:

> functions [...] may (behave (not (as expected)))

That means that "not" has no direct syntactical relation to "may", as indeed, English phrase "may not" is highly ambiguous (same applies to its past tense, thought indeed, connotations are different). If you want a second opinion on that refer to the list of formally defined modals from https://www.ietf.org/rfc/rfc2119.txt . While it includes "must" vs "must not" and "should" vs "should not", it includes only "may". The reason for that is what we discussed above - any phrase with "may not" is inherently ambiguous, so any doc faithful to RFC2119 would never put those 2 words in a row. And MicroPython docs try to follow these best practices.

The bottom line which would be nice for any contributor to keep in mind: many things in MicroPython are done for a reason. Not all, like there's no good reason in typos and missing articles, yet many things definitely have a reason behind them. Telling one from another may be not easy, but possible (and then it's better to err on fixing the obvious simple things rather than "everything at once").